### PR TITLE
Batch client get df

### PR DESCRIPTION
**Pull request by nemozen**
Creates the same data frame as the non-async GroClient but using async data retrieval for better performance.

### CODE REVIEW

File: api/client/batch_client.py [hunk 1]
```diff
@@ -84,11 +84,13 @@ def get_data_points(self, **selection):
         of: item_id, metric_id, region_id, frequency_id, source_id,
         partner_region_id. Additional arguments are allowed and ignored.
         """
-        headers = {'authorization': 'Bearer ' + self.access_token}
-        url = '/'.join(['https:', '', self.api_host, 'v2/data'])
-        params = lib.get_data_call_params(**selection)
-        resp = yield self.get_data(url, headers, params)
-        raise gen.Return(json_decode(resp))
+        data_points = super(BatchClient, self).get_data_points(**selection)
```
Comment (tolothy) [inline api/client/batch_client.py:87]: i think we are missing a yield here?

Comment (nemozen) [reply api/client/batch_client.py:87]: actually  yield would be if it was calling a coroutine. Here we are calling non-async get_data_points
https://www.tornadoweb.org/en/stable/gen.html#decorators

Comment (nemozen) [reply api/client/batch_client.py:87]: doign it both ways no as discussed offline

File: api/client/batch_client.py [hunk 2]
```diff
@@ -79,22 +79,33 @@ def get_data(self, url, headers, params=None):
...
+    @gen.coroutine
+    def get_data_points(self, **selection):
+        """Get all the data points for a given selection, which is some or all
+        of: item_id, metric_id, region_id, frequency_id, source_id,
+        partner_region_id. Additional arguments are allowed and ignored.
+        """
+        data_points = super(BatchClient, self).get_data_points(**selection)
+        raise gen.Return(data_points)
+
+    def get_df(self):
```
Comment (eliaschneider) [inline api/client/batch_client.py]: here we need add option show_revision=True

Comment (nemozen) [reply api/client/batch_client.py]: done

**Review by tolothy — APPROVED** [review]
tested `batch_async_get_data_points`

**Pull request merged**